pkg/operator/config/config.go

```diff
@@ -50,6 +55,12 @@ type OperatorControllerConfig struct {
 	// List of hooks should be run on the storage classes.
 	// No informers here, because StorageClassController does not accept any.
 	StorageClassHooks []csistorageclasscontroller.StorageClassHookFunc
+
+	// ExtraReplacements defines additional replacements that should be made to assets
+	ExtraReplacementsFunc func() []string
+
+	// SnapshotAssetName provides a way to override snapshot asset name that will be used
+	SnapshotAssetNameFunc func() []string
```

Comment:
I think VolumeSnapshotClasses and StorageClasses should have the same handling. I.e. there should be  `VolumeSnapshotClassHooks` that would set ASH specific parameters and not a separate yaml file. I know, it implies a whole new controller. Maybe it would be better in the long run than the StaticResourceController that we have now.

Reply:
Isn't less code better? :-) 

Seriously though - if all storageclass controller does is, performs cosmetic changes to storageclass, then it is perhaps a mistake to carry 300 lines of code (and tests), just for tweaking an parameter. Not to mention, a whole new controller results in additional api-usage, events, conditions and what not.

Reply:
We may need the same defaulting logic that we have in StorageClassController in volume snapshot too. My point is, storage classes and snapshot classes are +/- the same objects and thus should have the same behavior (they don't have that now) and code.
(Would there be a way to have a common ClassController?)

Reply:
Ok, this is actually the biggest issue in this PR (I mean the most complicated to fix). I'm fine with a TODO comment and let's fix it later.